### PR TITLE
Sorry for such a large pull request

### PR DESCRIPTION
**Pull request by tsvi**
It fixes most of #3 and some of #2 

There are here a bunch of styling fixes based on pylint and pydocstyle

There are here a bunch of unittests, not enough but it generates about 35% coverage

The main issue pylint is still complaining about, are too-many issues.

Also note: I removed the original libhcal source code, as it has nothing to provide here. I think a link is enough.

### CODE REVIEW

File: hdate/hdate_string.py [hunk 1]
```diff
@@ -1,121 +1,124 @@
...
+    omer_string += 'לעומר'
+    return omer_string
 
 
 def get_parashe(parasha, short=False, hebrew=True):
+    """Get the string representing the parasha."""
     is_hebrew = 1 if hebrew else 0
     is_short = 1 if short else 0
-    res = parashaot[is_hebrew]
+    res = PARASHAOT[is_hebrew][parasha]
```
Comment (royir) [inline hdate/hdate_string.py]: Did you found this bug using unit test?

File: tests/test_hdate.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import pytest
+import hdate
+import hdate.hdate_julian as hj
+
+import datetime
+
+
+HEBREW_YEARS_INFO = {
+    # year, dow rosh hashana, length, dow pesach
+    5753: (2, 353, 3), 5773: (2, 353, 3), 5777: (2, 353, 3),
```
Comment (royir) [inline tests/test_hdate.py]: did you compared the data to "external" source?

File: hdate/hdate_string.py (hunk 1 above)
Comment (tsvi) [inline hdate/hdate_string.py]: This was found by running pylint which complained that the variable was not in use

File: tests/test_hdate.py (hunk 2 above)
Comment (tsvi) [inline tests/test_hdate.py]: Yes, these are values provided for the various years in wikipedia

**Pull request merged**